skipper.go

```diff
@@ -1294,6 +1296,49 @@ func getRedisUpdaterFunc(namespace, name string, kdc *kubernetes.Client) func()
 	}
 }
 
+type RedisEndpoint struct {
+	Url string `json:"url"`
+}
+
+type RedisEndpoints struct {
+	Endpoints []RedisEndpoint `json:"endpoints"`
+}
+
+func updateEndpointsFromURL(address string) func() []string {
+	return func() []string {
+		/* #nosec */
+		resp, err := http.Get(address)
+		if err != nil {
+			log.Errorf("failed to connect to redis endpoint %v, due to: %v", address, err)
+			return nil
+		}
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.Errorf("failed to read to redis response %v", err)
+			return nil
+		}
+
+		target := &RedisEndpoints{}
+
+		err = json.Unmarshal(body, target)
+		if err != nil {
+			log.Errorf("Failed to decode body to json %v", err)
+			return nil
+		}
+
+		a := make([]string, 0, len(target.Endpoints))
+		for _, endpoint := range target.Endpoints {
+			a = append(a, endpoint.Url)
+		}
+
+		log.Infof("Redis updater called and found %d redis endpoints", len(a))
+
+		return a
```

Comment:
in case target.Endpoints=nil because of an error to reach the remote endpoints, we need to check that we don't change anything in net.RedisClient or wherever we use the func returned from updateEndpointsFromURL(). IIRC this is passed to redisclient UpdateFunc or whatever the name was.
What we need to omit is that we turn to 0 redis instances in case of an error.

Reply:
perfect, thanks!